pkg/asset/machines/openstack/machines.go

```diff
@@ -160,6 +157,7 @@ func generateProvider(clusterID string, platform *openstack.Platform, mpool *ope
 		PrimarySubnet:    platform.MachinesSubnet,
 		AvailabilityZone: az,
 		SecurityGroups:   securityGroups,
+		ServerGroupName:  clusterID + "-" + role + "-" + az,
```

Comment:
are we sure we want `az` here? what if it's empty? do we end up with `cluster123-worker-` ?

Reply:
Leveraged the power of neural networks, PTAL

Reply:
it was ugly indeed.